java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java

```diff
@@ -0,0 +1,67 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.stream;
+
+import glide.api.commands.StreamBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Builder;
+
+/**
+ * Optional arguments for {@link StreamBaseCommands#xgroupCreate(String, String, String,
+ * StreamGroupOptions)}
+ *
+ * @see <a href="https://valkey.io/commands/xgroup-create/">valkey.io</a>
+ */
+@Builder
+public final class StreamGroupOptions {
+
+    // Redis API String argument for makeStream
+    public static final String MAKE_STREAM_REDIS_API = "MKSTREAM";
+
+    // Redis API String argument for entriesRead
+    public static final String ENTRIES_READ_REDIS_API = "ENTRIESREAD";
+
+    /**
+     * If <code>true</code> and the stream doesn't exist, creates a new stream with a length of <code>
+     * 0</code>.
+     */
+    @Builder.Default private boolean mkStream = false;
+
+    public static class StreamGroupOptionsBuilder {
```

Comment:
Why do you need 2 builders? lines 30 and 15

Reply:
This will be the same builder.  We're writing this infrastructure to add the `makeStream()` signature (without boolean).

Reply:
this means we can do StreamGroupOptions.builder().makeStream().build();

Reply:
It is confusing, but "It is not silly if it works" ©

Reply:
Lombok is the best